src/main/scala/io/scalac/wtf/domain/UserService.scala

```diff
@@ -1,37 +1,18 @@
 package io.scalac.wtf.domain
 
-import io.scalac.wtf.domain.User.{UserAlreadyExists, ValidationError}
+import io.scalac.wtf.domain.User.ValidationError
+import cats.data.{NonEmptyList, Xor}
 import slick.dbio.DBIO
-import slick.driver.H2Driver.api._
-
-import cats.data.Validated.{invalidNel, valid}
-import cats.data.{NonEmptyList, ValidatedNel}
-import cats.implicits._
-import slick.lifted.MappedTo
 
 import scala.concurrent.ExecutionContext
 
-object UserService {
-
-  import scala.concurrent.ExecutionContext.Implicits.global
 
-  /*
-  //TODO: Move this somewhere else
-  //Evidence for DBIO being a cats Monad
-  implicit val dbioMonad = new Functor[DBIO] with Applicative[DBIO] with Monad[DBIO] {
-    def pure[A](a: A): DBIO[A] = DBIO.successful(a)
-    def flatMap[A, B](fa: DBIO[A])(f: A => DBIO[B]) = fa.flatMap(f)
-  }
-*/
-  def createUser(createdUser: User)(implicit executionContext: ExecutionContext): DBIO[ValidatedNel[ValidationError, UserId]] = {
-    val validation = User.validateUser(createdUser.email, createdUser.password)
+object UserService {
 
-    UserRepository.findByEmail(createdUser.email).flatMap {
-      case Some(_) => {
-        val userExistsValidated: ValidatedNel[ValidationError, UserId] = invalidNel(UserAlreadyExists)
-        DBIO.successful((validation |@| userExistsValidated) map { case (user, userId) => userId})
-      }
-      case None => UserRepository.save(createdUser).map(userId => (validation |@| valid(userId)) map { case (user, userId) => userId })
-    }
-  }
-}
\ No newline at end of file
+  def createUser(createdUser: User)(implicit ec: ExecutionContext): DBIO[Xor[NonEmptyList[ValidationError], UserId]] =
+    for {
+      validatedUser <- User.validateUser(createdUser.email, createdUser.password)
```

Comment:
Maybe use validatedUser in the next operation (save) or assign this to _? It would be IMO more explicit if You're not using it anywhere